src/script/descriptor.cpp

```diff
@@ -1418,24 +1425,34 @@ class TRDescriptor final : public DescriptorImpl
     }
     bool ToStringSubScriptHelper(const SigningProvider* arg, std::string& ret, const StringType type, const DescriptorCache* cache = nullptr) const override
     {
-        if (m_depths.empty()) return true;
+        if (m_depths.empty()) {
+            // If there are no sub-descriptors and a PRIVATE string
+            // is requested, return `false` to indicate that the presence
+            // of a private key depends solely on the internal key, not on
+            // any sub-descriptor. This ensures correct behavior for
```

Comment:
In deed439bff638abfc298a3b9c4f3d9f75fa18244 "descriptor: ToPrivateString() pass if  at least 1 priv key exists"

Micro nit if retouched

```diff
-            // of a private key depends solely on the internal key, not on
-            // any sub-descriptor. This ensures correct behavior for
+           // of a private key depends solely on the internal key (which is checked
+           // in the caller), not on any sub-descriptor. This ensures correct behavior for
```

Reply:
Done